app/middleware.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from django.conf import settings
+from django.http import HttpRequest
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.functional import SimpleLazyObject
+from praw import Reddit
+from social_django.utils import load_strategy
+
+log = logging.getLogger(__name__)
+
+
+def refresh_token(request: HttpRequest) -> None:
+    """Refresh the OAuth token so that we can use it."""
+    strategy = load_strategy(request)
+    social = request.user.social_auth.filter(provider="reddit")[0]
+    social.refresh_token(
+        strategy=strategy, redirect_uri="http://localhost:8000/complete/reddit/"
+    )
+
+
+def configure_reddit(request: HttpRequest) -> Reddit:
+    """Build the reddit instance that will be attached to the user."""
+    reddit = Reddit(
+        client_id=settings.SOCIAL_AUTH_REDDIT_KEY,
+        client_secret=settings.SOCIAL_AUTH_REDDIT_SECRET,
+        refresh_token=request.user.social_auth.first().extra_data["refresh_token"],
+        user_agent=(
+            f"blossomtest, logged in as {request.user.username};"
+            f" contact u/itsthejoker with questions."
+        ),
```

Comment:
I think it would be good here to follow the `<platform>:<app ID>:<version string> (by /u/<reddit username>)` naming scheme as specified in the [Reddit API docs](https://github.com/reddit-archive/reddit/wiki/API)

Reply:
Yes... but we still need to change it on a per-logged-in-user basis. Even if different people are logged in, the user agent gate is still up, so there's the 600 requests / 10 minute window that we have to get around. I'll update it to the requested format, though.